compiler/rustc_lint/src/unqualified_local_imports.rs

```diff
@@ -0,0 +1,83 @@
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::{self as hir};
+use rustc_session::{declare_lint, declare_lint_pass};
+use rustc_span::symbol::kw;
+
+use crate::{lints, LateContext, LateLintPass, LintContext};
+
+declare_lint! {
+    /// The `unqualified_local_imports` lint checks for `use` items that import a local item using a
+    /// path that does not start with `self::`, `super::`, or `crate::`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,edition2018
+    /// #![warn(unqualified_local_imports)]
+    ///
+    /// mod localmod {
+    ///     pub struct S;
+    /// }
+    ///
+    /// use localmod::S;
+    // We have to actully use `S`, or else the `unused` warnings supress the lint we care about.
```

Comment:
```suggestion
    // We have to actually use `S`, or else the `unused` warnings suppress the lint we care about.
```

Reply:
A `//` comment in the middle of a doc comment looks odd. Maybe precede this comment with `/// # `?

Reply:
Then it would have to be `/// # //`... what a symbol soup.^^ But I can do that.

We use `//` a couple times in std/core to add FIXMEs to doc comments. But here the comment actually refers to the code below that us under `/// #` so it makes sense to also use that for the comment.